frappe/integrations/utils.py

```diff
@@ -19,7 +19,10 @@ def make_request(method, url, auth=None, headers=None, data=None):
 		s = get_request_session()
 		frappe.flags.integration_request = s.request(method, url, data=data, auth=auth, headers=headers)
 		frappe.flags.integration_request.raise_for_status()
-
+		
+		if return_response_obj :
+			return frappe.flags.integration_request
```

Comment:
This is overall a very hacky API. 

It returns a Response object if one flag is passed and parsed data if the content type matches to some hardcoded value? 🤦  


Looks like this was never designed to be generic enough... maybe better to create a new API?

Reply:
Ideally, the return type of functions should be fairly consistent. 

This version now returns apples and oranges.

Reply:
Hello @ankush,
Sorry for the late response. Currently I did not find a way to play with external APIs in Server Script and the `make_request()` only succeeds if the return content type is `text/plain; charset=utf-8` or json. it fails when content type is in some other format or a blob. Also, there is no way to build complex logics by utilizing respnse code, that's why I think adding an option to get the entire Response Object will be an advantage.


I am new here, so I have less understanding.

Reply:
All I am proposing is create a separate function that returns response by default instead of processing it. 

We can verify the security concerns with it separately. 


Merging this change might solve your problem but make API worse than it already is. 


Reopen if you're interested.